update: adding information about new context-sensitive resolution (#4865)
JetBrains/kotlin-web-site: docs/topics/control-flow.md

}
```

> > To simplify your `when` expressions and reduce repetition, you can try out context-sensitive resolution (currently in preview).

Author: sure, I like it and it works still — makes it more concise 👍

> > To simplify your `when` expressions and reduce repetition, you can try out the Experimental context-sensitive resolution feature.

Reviewer: ```suggestion
> To simplify your `when` expressions and reduce repetition, you can try out context-sensitive resolution (currently in preview).
```

I think the right wording here is "preview" or "beta", not "experimental"
